src/v/datalake/tests/record_generator.cc

```diff
@@ -52,7 +52,7 @@ record_generator::add_random_avro_record(
         co_return error{fmt::format("Schema {} is missing", name)};
     }
     auto schema_id = it->second;
-    auto schema_def = co_await _sr->get_valid_schema(schema_id);
+    auto schema_def = *co_await _sr->get_valid_schema(schema_id);
```

Comment:
no check needed before dereferencing the optional? or is this caught by the upper layers?

Reply:
Lazy test code :) will fix